shotgun_api3/lib/mockgun/mockgun.py

```diff
@@ -440,22 +447,44 @@ def _validate_entity_data(self, entity_type, data):
-                    raise ShotgunError("%s.%s is of type multi_entity, but data %s contains a non-dictionary" % (entity_type, field, item))
+                    raise ShotgunError(
+                        "%s.%s is of type multi_entity, but data %s contains a non-dictionary" %
+                        (entity_type, field, item)
+                    )
                 elif item and any("id" not in sub_item or "type" not in sub_item for sub_item in item):
-                    raise ShotgunError("%s.%s is of type multi-entity, but an item in data %s does not contain 'type' and 'id'" % (entity_type, field, item))
-                elif item and any(sub_item["type"] not in field_info["properties"]["valid_types"]["value"] for sub_item in item):
-                    raise ShotgunError("%s.%s is of multi-type entity, but an item in data %s has an invalid type (expected one of %s)" % (entity_type, field, item, field_info["properties"]["valid_types"]["value"]))
-
+                    raise ShotgunError(
+                        "%s.%s is of type multi-entity, but an item in data %s does not contain 'type' and 'id'" %
+                        (entity_type, field, item)
+                    )
+                elif item and any(
+                    sub_item["type"] not in field_info["properties"]["valid_types"]["value"] for sub_item in item
+                ):
+                    raise ShotgunError(
+                        "%s.%s is of multi-type entity, but an item in data %s has an invalid type (expected one of %s)"
+                        % (entity_type, field, item, field_info["properties"]["valid_types"]["value"])
+                    )
 
             elif field_info["data_type"]["value"] == "entity":
                 if not isinstance(item, dict):
-                    raise ShotgunError("%s.%s is of type entity, but data %s is not a dictionary" % (entity_type, field, item))
+                    raise ShotgunError(
+                        "%s.%s is of type entity, but data %s is not a dictionary" %
+                        (entity_type, field, item)
+                    )
                 elif "id" not in item or "type" not in item:
-                    raise ShotgunError("%s.%s is of type entity, but data %s does not contain 'type' and 'id'" % (entity_type, field, item))
-                #elif item["type"] not in field_info["properties"]["valid_types"]["value"]:
-                #    raise ShotgunError("%s.%s is of type entity, but data %s has an invalid type (expected one of %s)" % (entity_type, field, item, field_info["properties"]["valid_types"]["value"]))
+                    raise ShotgunError(
+                        "%s.%s is of type entity, but data %s does not contain 'type' and 'id'"
+                        % (entity_type, field, item)
+                    )
+                # elif item["type"] not in field_info["properties"]["valid_types"]["value"]:
```

Comment:
any reason why this is commented out? maybe make a comment to say why it has been left in?

Reply:
Maybe it's not obvious from the diff, but it was already commented out. I only reformatted it because it was longer than 120 chars. Unfortunately we don't know why it was commented out...